chevah/brink
# [#4905] Update Windows dev distribution for git-lfs and store buildbot credentials in external files.

Scope
=====

This is an update for the Windows dev environment to have rsync and git-lfs.

Changes
=======

Create a new archive with git-lfs and rsync.

Add documentation to the archive (version file, readme files)

Simplify the msys-console.js script.

This also adds support for sha256 checksum as an update to md5.

How to try and test the changes
===============================

reviewers: @lgheorghiu 

@lgheorghiu you should download the msys-console.js script from this branch and run it.
It should create a new dev environment which has git-lfs and rsync and ssh-rsync

@lgheorghiu you should uninstall git-lfs if you have it manually installed.

Let me know if other tools needs to be installed.

---------

@dumol please check that paver.conf is ok... is the same as in compat now :) 

I added as a reviewer just to know that you Windows base env was updated.
You can give it a try on Windows VM, or just wait for when a new Win VM can be created.
The git version is the same, this just adds extra tools.


1.Publish to staging from Windows
----------------------------------

In a branch on chevah/server or sftpplus.com modify paver.conf to have `chevah-brink==0.69.2`
Then run paver.deps


You will need to create `c:\users\YOUR-USER\.ssh\config` file and add Port for the staging.sftpplus.com as documented in the README from sftpplus.com

the content of `.ssh\config` for Windows
```
UserKnownHostsFile c:\Users\YOUR-USER\.ssh\known_hosts
Host staging.sftpplus.com
    Port 12022
    IdentityFile C:\Users\YOUR-USER\.ssh\id_rsa
```

`C:\Users\YOUR-USER\.ssh\id_rsa` is the path to the SSH key which you configured to allow uploads.

Then try `paver publish --target=staging` on a SFTPPlus.com branch or `paver publish_documentation` on chevah/server branch.
